tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -2317,4 +2317,121 @@ public function userGetsThePersonalDriveInformationOfUserUsingGraphApi(string $b
+		$userId = $this->featureContext->getAttributeOfCreatedUser($ofUser, 'id') ?? $ofUser;
+		$credentials = $this->getAdminOrUserCredentials($user);
+
+		$response = GraphHelper::getAssignedRole(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$this->featureContext->getAdminUsername(),
+			$this->featureContext->getAdminPassword(),
+			$userId
+		);
+		$appRoleAssignmentId = $this->featureContext->getJsonDecodedResponse($response)["value"][0]["id"];
+
+		$this->featureContext->setResponse(
+			GraphHelper::unassignRole(
+				$this->featureContext->getBaseUrl(),
+				$this->featureContext->getStepLineRef(),
+				$credentials['username'],
+				$credentials['password'],
+				$appRoleAssignmentId,
+				$userId
+			)
+		);
+	}
+
+	/**
+	 * @Then user :user should have the role :role assigned
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userShouldHaveTheRoleAssigned(string $user, string $role): void {
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id') ?? $user;
+		$response = GraphHelper::getAssignedRole(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$this->featureContext->getAdminUserName(),
+			$this->featureContext->getAdminPassword(),
+			$userId
+		);
+
+		$jsonDecodedResponse = $this->featureContext->getJsonDecodedResponse($response)['value'][0];
+		if (empty($this->appEntity)) {
+			$this->setApplicationEntity();
+		}
+		Assert::assertEquals(
+			$this->appEntity["appRoles"][$role],
+			$jsonDecodedResponse['appRoleId'],
+			__METHOD__
+			. "\nExpected role ID for role '$role'' to be '" . $this->appEntity["appRoles"][$role] . "' but got '" . $jsonDecodedResponse['appRoleId'] . "'"
+		);
+	}
+
+	/**
+	 * @Then user :user should not be able to create a group :group
+	 *
+	 * @param string $user
+	 * @param string $group
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userShouldNotBeAbleToCreateAGroup(string $user, string $group): void {
+		$response = $this->createGroup($group, $user);
+
+		$actualStatusCode = $response->getStatusCode();
+		Assert::assertEquals(
+			403,
+			$actualStatusCode,
+			"HTTP status code $actualStatusCode is not one of the expected values 403"
+		);
+	}
+
+	/**
+	 * @Then user :user should not have any role assigned
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userShouldNotHaveAnyRoleAssigned(string $user): void {
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id') ?? $user;
+		$response = GraphHelper::getAssignedRole(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$this->featureContext->getAdminUserName(),
+			$this->featureContext->getAdminPassword(),
+			$userId
+		);
```

Comment:
same here ^

Reply:
There is no need to use appEntity, as the objective of the step is to check whether the assigned role is empty.